### PR TITLE
Master audio integration

### PR DESCRIPTION
**Pull request by CobbieCobbie**
MasterAudio Integration is finished
- Music & SFX slider functionality enabled
- Mixing of the sfx
- Custom events fired with code triggers. Two PlaylistControllers are running simultaneously for smooth fades, event triggered when switching scenes and playlistcontrollers are initialized on MasterAudio side, not Unity game object side
- Adapting the Music Design to the Game Design

### CODE REVIEW

File: Assets/AudioObject.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AudioObject : MonoBehaviour
+{
+    // Start is called before the first frame update
```
Comment (locojoetive) [inline Assets/AudioObject.cs:7]: Remove Start and Update methods!

File: Assets/_prefabs/GameMode/GameMode.cs [hunk 2]
```diff
@@ -72,11 +86,14 @@ private void OnChangeBallSpeed(System.Single _ballSpeed)
     private void OnChangeSfxVolume(System.Single _sfxVolume)
     {
         sfxVolume = (int)_sfxVolume;
+        MasterAudio.MasterVolumeLevel = (float)_sfxVolume / 20f;
+        //MasterAudio.SetBusVolumeByName("Ball_SFX", (float)_sfxVolume / 20f);
+        //MasterAudio.SetBusVolumeByName("Human_SFX", (float)_sfxVolume / 20f);
```
Comment (locojoetive) [inline Assets/_prefabs/GameMode/GameMode.cs:91]: remove comments

File: Assets/_prefabs/GameMode/GameMode.cs [hunk 3]
```diff
@@ -72,11 +86,14 @@ private void OnChangeBallSpeed(System.Single _ballSpeed)
...
     }
     private void OnChangeMusicVolume(System.Single _musicVolume)
     {
         musicVolume = (int)_musicVolume;
-        // TODO: Set Music Volume here
+        MasterAudio.PlaylistMasterVolume = (float) _musicVolume / 20f;
```
Comment (locojoetive) [inline Assets/_prefabs/GameMode/GameMode.cs:96]: add comment to magic number

File: Assets/_prefabs/InGame/Handlers/_scripts/StageManager/StageManager.cs [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
 using System;
+using DarkTonic.MasterAudio;
 
 public class StageManager : MonoBehaviour
 {
     private void Start()
     {
         Time.timeScale = 1f;
-        // MUSIC: Start Theme
     }
```
Comment (locojoetive) [inline Assets/_prefabs/InGame/Handlers/_scripts/StageManager/StageManager.cs:10]: where is volume of the two playlist controllers changed?

File: Assets/_prefabs/UI/TitleHandler.cs [hunk 5]
```diff
@@ -1,5 +1,6 @@
 using UnityEngine;
 using UnityEngine.SceneManagement;
+using DarkTonic.MasterAudio;
```
Comment (locojoetive) [inline Assets/_prefabs/UI/TitleHandler.cs:3]: remove import if it is not used

**Review by locojoetive — COMMENTED** [review]
some minor changes needed

File: Assets/_prefabs/InGame/Handlers/_scripts/StageManager/StageManager.cs (hunk 4 above)
Comment (CobbieCobbie) [reply Assets/_prefabs/InGame/Handlers/_scripts/StageManager/StageManager.cs:10]: what do you mean?

**Review by locojoetive — APPROVED** [review]
all good

**Review by herrnock — APPROVED** [review]

**Pull request merged**